src/interfaces/widgets/module.py

```diff
@@ -0,0 +1,300 @@
+"""
+
+    This module generate serveral combination frame based object.
+
+"""
+
+from PyQt5.QtWidgets import QWidget
+from PyQt5.QtCore import Qt
+
+from .button import Button, RadioButton
+from .frame import Frame
+from .groupbox import GroupBox
+from .label import Label
+from .layout import HorizontalLayout
+from .lineedit import LineEdit
+from .spacer import HorizontalSpacer
+
+
+class BaseTwoLabelFrame(Frame):
+
+    label_one : Label = None
+    label_two : Label = None
+
+    def __init__(self, widget, **kwargs):
+        super(BaseTwoLabelFrame, self).__init__(widget, **kwargs)
+
+        kwargs.pop('name', None)
+        get_param = lambda x : kwargs.get(x)
+
+        label_one_name = get_param("label_one_name")
+        label_two_name = get_param("label_two_name")
+
+        label_one_text = get_param("label_one_text")
+        label_two_text = get_param("label_two_text")
+
+        layout = HorizontalLayout(self)
+
+        self.label_one = Label(self, name=label_one_name, **kwargs)
+        self.set_label_one_text(label_one_text)
+        layout.addWidget(self.label_one)
+
+        spacer = HorizontalSpacer()
+        layout.addItem(spacer)
+
+        self.label_two = Label(self, name=label_two_name, **kwargs)
+        self.set_label_two_text(label_two_text)
+        layout.addWidget(self.label_two)
+
+    def set_label_one_text(self, text:str):
+        self.label_one.setText(text)
+
+    def set_label_two_text(self, text:str):
+        self.label_two.setText(text)
+
+    def get_label_one(self):
+        return self.label_one
+
+    def get_label_two(self):
+        return self.label_two
+
+
+class SectionTitleFrame(BaseTwoLabelFrame):
+
+    def __init__(self, widget, **kwargs):
+        super(SectionTitleFrame, self).__init__(widget,
+                                                name="section_title_frame",
+                                                label_one_name="section_title",
+                                                label_two_name="section_hint",
+                                                align=Qt.AlignVCenter, **kwargs)
+
+
+class ConfigFrame(BaseTwoLabelFrame):
+
+    def __init__(self, widget, **kwargs):
+        super(ConfigFrame, self).__init__(widget, name="config_frame",
+                                          align=Qt.AlignVCenter, **kwargs)
+
+    def setTitle(self, text:str):
+        super().set_label_one_text(text)
+
+    def title(self):
+        return self.label_one.text()
+
+    def setText(self, text:str):
+        super().set_label_two_text(text)
+
+    def text(self):
+        return self.label_two.text()
+
+    def green(self):
+        self.setObjectName("config_frame_green")
+
+    def red(self):
+        self.setObjectName("config_frame_red")
+
+    def reset(self):
+        super().set_label_two_text("-")
+        self.setObjectName("config_frame")
+
+
+class BaseInputFrame(Frame):
+
+    title       : Label = None
```

Comment:
Is there a reason these are defined as class variables instead of instance variables (i.e. within `__init__`)

Reply:
instance variables and class variables performs mostly the same. One of the difference is that when we try to access an attribute on an instance, it first checks whether the instance contains the attribute. In this case, we want outer scope has access to the child widget of `BaseInputFrame`, so I leave it to be class variables.